### DISCUSSION TITLE
Is the "From Template" option available for Quarto documents like R Markdown?

### DISCUSSION THREAD

**Discussion by danielludolf** [quarto-dev/quarto-cli#2079] [Q&A]
I have built R packages in the past that include custom R Markdown templates that render to PDF or HTML, but I did not see the option initially available for Quarto documents when in RStudio. Is creating a R package that includes custom templates an option for Quarto? If so, are there any reference materials on how this is possible?

**Comment by maxdrohde** [comment]
This may be an example of what you're looking for?

https://github.com/quarto-journals/

**Comment by danielludolf** [comment]
@maxdrohde The link you provided is helpful, but I was more curious if [Chapter 17 from R Markdown: The Definitive Guide](https://bookdown.org/yihui/rmarkdown/document-templates.html)'s capabilities were available within Quarto? The templates I have created are custom company templates, and I do not necessarily have an interest in creating new templates based on journals.

**Reply by dragonstyle** [reply]
In Quarto, we don't yet have the nice integration for creating a new file or project from a template, but we have the raw materials in the form of format extensions. These allow you to author reusable formats (e.g. mycorp-pdf) that would bundle the styling and so on that you would like. Users can install these format extensions and then just target that format to use it when rendering a document.

See https://quarto.org/docs/extensions/formats.html#format-templates for more about this.

**Comment by mcanouil** [comment]
I suggest to take a look at a quick demo made by @jthomasmock <https://github.com/jthomasmock/octavo> to embed a Quarto extension in an R package (source: <https://twitter.com/thomas_mock/status/1558919681263689733>)
To note, several alternatives are possible.

As for RMarkdown, files can be added into `inst` directory, and one can write a function to copy the files in the current Quarto project.
For now, the r package `quarto` can't install extension, I made a PR to add this feature (https://github.com/quarto-dev/quarto-r/pull/45). In theory, you could then use it to install/use an extension/format from within any R package (maybe I'll have to tweak a bit the functions in the PR, or at least test for this use case).
